### PR TITLE
[v1] Webhooks

### PR DESCRIPTION
**Pull request by rafaelsideguide**

### CODE REVIEW

**Comment by nickscamara** [comment]
- [ ] Need docs

File: apps/api/src/services/webhook.ts [hunk 1]
```diff
@@ -50,7 +50,7 @@ export const callWebhook = async (teamId: string, jobId: string, data: any, spec
       },
       body: JSON.stringify({
         success: data.success,
-        jobId: jobId,
+        id: id,
```
Comment (nickscamara) [inline apps/api/src/services/webhook.ts]: cant change this for v0

File: apps/api/src/services/queue-worker.ts [hunk 2]
```diff
@@ -341,11 +347,20 @@ async function processJob(job: Job, token: string) {
...
-
-        await callWebhook(job.data.team_id, job.data.crawl_id, data, job.data.webhook, job.data.v1);
+          // v0 web hooks, call when done with all the data
+        if (!job.data.v1) {
+          callWebhook(job.data.team_id, job.data.crawl_id, data, job.data.webhook, job.data.v1, "crawl.completed");
+        }
+        // v1 web hooks, call when done with no data, but with event completed
+        if (job.data.v1 && job.data.webhook) {
+          callWebhook(job.data.team_id, job.data.crawl_id, [], job.data.webhook, job.data.v1, "crawl.completed");
+        }
```
Comment (mogery) [inline apps/api/src/services/queue-worker.ts]: do this before gathering all data to spare server RAM

**Review by mogery — CHANGES_REQUESTED** [review]
lgtm other than that one thing. it's a serious issue

File: apps/api/src/services/queue-worker.ts (hunk 2 above)
Comment (nickscamara) [reply apps/api/src/services/queue-worker.ts]: fixed

**Pull request merged**